File: frontend/src/core/validators/highFrequencySmallInterval.js
export function validate(store, transactionData) {
  const account = store.getAccount();
  if (account === null) {
    return [];
  }

  const currentTime = transactionData.time;
  if (!currentTime) {
    return [];
  }

  const twoMinutesAgo = new Date(currentTime.getTime() - 2 * 60 * 1000);

  const recentTransactions = store.getTransactions().filter(
    t => t.time >= twoMinutesAgo
  );

  if (recentTransactions.length >= 3) {
    return ['high-frequency-small-interval'];
  }
  return [];
}

File: backend/src/validators/high_frequency_small_interval.py
from datetime import timedelta
from ..core.state import Store


def validate(store: Store, transaction_data: dict) -> list[str]:
    account = store.get_account()
    if account is None:
        return []
    
    current_time = transaction_data.get("time")
    if current_time is None:
        return []
    
    two_minutes_ago = current_time - timedelta(minutes=2)
    
    recent_transactions = [
        t for t in store.get_transactions()
        if t.time >= two_minutes_ago
    ]
    
    if len(recent_transactions) >= 3:
        return ["high-frequency-small-interval"]
    return []